velox/functions/sparksql/Arithmetic.h

```diff
@@ -318,4 +321,53 @@ struct IsNanFunction {
     }
   }
 };
+
+template <typename T>
+struct ToHexVarbinaryFunction {
```

Comment:
I think you can combine all 3 structs into one: ToHexFunction

and have 3 call methods with different input types

Reply:
done, thanks!

Reply:
I still see 3 different structs.

Reply:
I got your point, but how can we define `call(const arg_type<Varbinary>& input)` and `call(const arg_type<Varchar>& input)` in one struct? 

I remember we have discussed before in https://github.com/facebookincubator/velox/pull/5042#discussion_r1227263826, compiler still report compile error now.

Reply:
Thank you for the reminder. I forgot about this limitation.